GROUPS.md

```diff
@@ -0,0 +1,43 @@
+## Introduction
+
+This page contains a few initial notes on observation group requirements and design.  The idea is to ensure that we understand what we're trying to accomplish before we spend too much time doing an implementation.
+
+## Group Concept
+
+In the existing OCS groups are explicitly part of the science program hierarchy, cannot be nested, and have limited impact on scheduling.  An observation can be in exactly zero or one group.  For OCS3, our idea is to capture groups by tagging observations (and other groups?).  This is the same concept behind "labels" in gmail.  What appear to be folders of emails are in fact just collections of emails with the same label.  Groups will work in a similar way in OCS3, except that there will be different types of group each with their own characteristics and scheduling implications.  It remains an open question how to portray groups in the user interface.  Like gmail, one can imagine a view of folders of observations for instance but folders probably won't capture timing constraints in an intuitive way.
+
+## Nested Groups
+
+To make more sophisticated scheduling constraints possible, it should be possible to tag a group itself with a group label.  Unfortunately "nesting" groups also invites the possibility of creating recursive groups.  "Group A" can be tagged with "Group B" and vice versa. It isn't clear what, if anything, should be done to prevent this other than to report any impossible scheduling constraints that result from nested groups.
+
```

Comment:
If we're going to allow nesting then I think the tagging notion may not be sufficient.